pkg/core/blockchain.go

```diff
@@ -1009,6 +1015,53 @@ func (bc *Blockchain) GetConfig() config.ProtocolConfiguration {
 	return bc.config
 }
 
+// CalculateClaimable calculates the amount of GAS which can be claimed for a transaction with value.
+func (bc *Blockchain) CalculateClaimable(value util.Fixed8, startHeight, endHeight uint32) (util.Fixed8, util.Fixed8, error) {
+	var amount util.Fixed8
+	di := uint32(bc.decrementInterval)
+
+	ustart := startHeight / di
+	if genSize := uint32(len(bc.generationAmount)); ustart < genSize {
+		uend := endHeight / di
+		iend := endHeight % di
+		if uend >= genSize {
+			uend = genSize - 1
+			iend = di
+		} else if iend == 0 {
+			uend--
+			iend = di
+		}
+
+		istart := startHeight % di
+		for ustart < uend {
+			amount += util.Fixed8(di-istart) * util.Fixed8(bc.generationAmount[ustart])
+			ustart++
+			istart = 0
+		}
+
+		amount += util.Fixed8(iend-istart) * util.Fixed8(bc.generationAmount[ustart])
+	}
+
+	var sysFeeTotal util.Fixed8
+	if startHeight == 0 {
+		startHeight++
+	}
+	for i := startHeight; i < endHeight; i++ {
+		h := bc.GetHeaderHash(int(i))
+		b, err := bc.GetBlock(h)
+		if err != nil {
+			return 0, 0, err
+		}
+		for _, tx := range b.Transactions {
+			sysFeeTotal += bc.SystemFee(tx)
```

Comment:
It's very expensive, I think we should save the system fee in `(*dao).StoreAsBlock`.

Reply:
done